rust_src/src/buffers.rs

```diff
@@ -159,6 +159,26 @@ impl LispBufferRef {
             self.fetch_byte(n) as c_int
         }
     }
+
+    #[inline]
+    pub fn overlays_before(&self) -> Option<LispOverlayRef> {
+        let ov = self.overlays_before;
+        if ov.is_null() {
```

Comment:
This looks like a pattern that might be worthwhile to pull out into its own function.

Reply:
`overlays_before` and `overlays_after` seem to be used in several places in the c code. Do you think it's enough to justify a method ?

Reply:
I am thinking that "is_null ? None : Some(transmute)" is probably a pattern.

Reply:
As it is that block is repeated 3 times in this file.

Reply:
`ptr.as_ref().map(mem::transmute)` should do it